### DISCUSSION TITLE
Ender 3 V2 Schematic for 4.2.2 board

### DISCUSSION THREAD

**Discussion by gothcha** [Jyers/Marlin#814] [General Discussion]
I have generated a schematic for the 4.2.2 pcb. It covers the TMC2208/9 drivers.

I am confident that it is pretty accurate, however some capacitor values could not be determined, but it should be helpful for us when discussing the various modifications.

Any input or corrections welcome.

_edited to include latest schematics_

[Creality 4.2.2 - Schematic 28-5-22.pdf](https://github.com/Jyers/Marlin/files/8791332/Creality.4.2.2.-.Schematic.28-5-22.pdf)
[Creality 4.2.7 - Schematic 28-5-22.pdf](https://github.com/Jyers/Marlin/files/8791333/Creality.4.2.7.-.Schematic.28-5-22.pdf)

**Comment by Tuncay-Ayhan** [comment]
You ma dude, are an Angel

**Reply by gothcha** [reply]
My pleasure, I was so frustrated guessing what was going on! It took a while to trace everything out!

**Comment by tiagofreire-pt** [comment]
Awesome job!

Are you sure the TMC2209 has the same wiring on 4.2.2 boards?

**Reply by gothcha** [reply]
The board itself is identical, so the wiring is the same. The 2209 has the Spread pin but the 2208 does not. Using UART gives more functionality than just enabling SpreadCycle mode, so I think this is the best and easiest option.

**Reply by Reggi3** [reply]
all the 4.2.2 board driver chips are pin compatible, drop in replacements, if you had a hot air gun you could relatively easily swap out chips.  The odd one out of the bunch is the 4.2.7 with TMC225s on it, I have seen that there is a 4.3.1 board as well.

**Comment by Reggi3** [comment]
Really nice work!

**Reply by gothcha** [reply]
Thank you @Reggi3 

I have made a few corrections, based on the up close photo of the TMC2209 by @msdossajo I now do not believe that pin 7 (SPREAD) is connected to R28

![TMC2209](https://user-images.githubusercontent.com/80574048/118249310-32cb9500-b49d-11eb-9838-eba150cb1768.jpg)

plus I shorted a 3.3V net to ground!.

**Comment by agalakhov** [comment]
Thank you sooo much!

**Comment by agalakhov** [comment]
Do tthermal sensor connectors TB and TH really have no pull-ups?

**Reply by gothcha** [reply]
Yes I have been adding these in, I just have to verify the values etc. Should have an update tomorrow.
Plus I need to add the components around the P3 pins.

**Reply by gothcha** [reply]
OK so this should now be complete - hopefully! Interestingly I have found that the pin P3 is another available input to the STM (PC0) which could be used for UART etc, it has a 10K pull-up built in.

<img width="345" alt="Screenshot 2021-05-15 at 11 14 38" src="https://user-images.githubusercontent.com/80574048/118356787-d5a01400-b56e-11eb-906c-c69b1d20e921.png">

**Reply by agalakhov** [reply]
Looks like R57 and R58 are not connected yet. And is the "A" pin of the LCD port really unconnected? It is probably an encoder phase signal.

**Reply by gothcha** [reply]
Oops I missed that. I have rechecked the LCD port and fixed it up. It is fiddly to beep out all the connections to the STM.

**Reply by gothcha** [reply]
@agalakhov Keep checking I am just running through all the component numbers to see if there are any present I have not accounted for ...

OK I believe that all the components are now accounted for, whew!

I have updated also the original post to contain the latest version.

[Ender 3V2 4.2.2 - Schematic 160521.pdf](https://github.com/Jyers/Marlin/files/6487013/Ender.3V2.4.2.2.-.Schematic.160521.pdf)

**Comment by dwery** [comment]
I see PC6 is connected to the display port, but unused by the display?

**Reply by gothcha** [reply]
Yes it looks that way based on the LCD pin out. But not easy to re-use on the connector.

**Reply by Reggi3** [reply]
what about tapping that pin off the back of the connector?

**Reply by dwery** [reply]
A little breakout should do, like the one that was used for the beeper pin. I see there are two pins for the buttons, but I think just one is used on the LCD board, right?

**Reply by gothcha** [reply]
The A & B are the encoder signals, both are required. ENT is the button. A breakout is a good solution.

**Reply by dwery** [reply]
I saw an ENT1 and ENT2 but was looking at the code -.-

**Comment by pmaersk** [comment]
Hi

> I have generated a schematic for the 4.2.2 pcb. It covers the TMC2208/9 drivers.

Thanks for sharing. This is really great. Any chance you would share the cad file for the schematics. Planning extending my 4.2.2 board with 2-3 more stepper controllers (despite soldering to the CPU will be a bitch) and using your drawing as basis for my documentation adding changes would be a really good start. Thanks in advance.

Peter Maersk-Moller

**Reply by gothcha** [reply]
Yes of course, it is generated with DesignSpark, what were you going to use?

**Reply by pmaersk** [reply]
Thanks a lot. Was/is planning using KiCAD for Linux/Mac. A conversation from 2017 says KiCAD can't read DesignSpark files. Perhaps situation has changed. Needs to check if situation has changed. Alternatively checking if DesignSpark can run on Mac/Linux and if a free version is available. There seems to be a free trial for DesignSpark Electrical, but that does not sound like free and it seems only to be available for Windows. Maybe you have a recommendation for another solution?
Best regards. Peter MM.

**Reply by gothcha** [reply]
No DesignSpark is PC only, I run it with Parallels on my Mac. I also use KiCad but prefer DS, in fact I started it in KiCad but wasn’t happy with the schematic entry and libraries, so went back to DS. I have not transferred between the two either

**Reply by gothcha** [reply]
[Creality 4.2.2.sch.zip](https://github.com/Jyers/Marlin/files/6651329/Creality.4.2.2.sch.zip)

The DS schematic file if you can use it.

**Reply by pmaersk** [reply]
Thanks a mill. Now I may have a starting point.
P

**Reply by gothcha** [reply]
@pmaersk Here is a copy of the updated schematic file with the error corrected

[Creality 4.2.2 16-6-21.sch.zip](https://github.com/Jyers/Marlin/files/6656704/Creality.4.2.2.16-6-21.sch.zip)

**Reply by pmaersk** [reply]
Thanks again.

**Comment by smooreace** [comment]
OMG it would be AMAZING to be able to add a second properly driven Z
controller! Ive been debating on swapping out to a BTT board, but the only
one that really fits the bill with dual Z and dual extruders is the
Octopus, and its pretty "old" when compared to the newer boards out there

**Reply by Reggi3** [reply]
extra steppers is pretty simple, just requires 2 pins (step, dir), EN pin should be connected to the same pin EN is connected to for all axis (PC3 iirc).  afaik the driver uses 3.3v for IO power which you can get from the SWDIO header, then you just need to route 24v and gnd from the 24v in screw terminals to the motor power pins on the driver chip.

**Reply by pmaersk** [reply]
I was planning using the UART mode for all stepper controllers. That should eliminate the need for the shared enable currently in use. As far as I figure it out, I need to connect the DIR and the UART from the stepper controller (TMC2208 in my case) to the ARM controller so using two ports per extra stepper controller. It does require some precise soldering unfortunately. I will test first on the SWDIO/SWCLK easy accessible. If it works, there should free ports enough for X,Y,Z1,E0 (the current setup) plus Z2, E1, E2, E3, E4 and still have bed leveling and filament runout and then have z-endstop plus the current shared enable port as a reserve.

As Reggi3 write I'll need 24V too. Plan is to use a second 24V PS for bed heating (moving it away from current PS and using as relay to enable it) so there will be enough power in the current PS of my Ender 3 V2. All these extra steppers will use roughly 10-15 watt each.

**Reply by pmaersk** [reply]
Hi Reggie.
If you are running the stepper controller in UART mode, you can skip the enable pin. But do you in UART mode connecting the UART pin, do you still need to connect both DIR and Step pin? Or only the UART and DIR pin? If needing 3 pins in UART mode (DIR, STep, UART), then of course I can't connect all the controllers I listed in mail above - sadly enough.

**Reply by Reggi3** [reply]
Hi, sorry I didn't reply earlier!  Yes it still needs Step and Dir pins with our current firmware, however it might be possible to use the internal driver stepping functions but I have no experience of that, I'm not sure if there is support in marlin for it.  There is another option that might help but it means losing the LCD connected directly to the printer but the LCD might still be usable via a pi and octoprint. 

The LCD connector uses 7 pins iirc, 2 for the serial connection, 3 for the encoder and 1 for the buzzer (6 are actually in use, the 7th is already spare), which would give you a lot more options for adding motors.

There is a python based solution for [adding the e3v2 LCD to a raspberry pi here](https://github.com/odwdinc/DWIN_T5UIC1_LCD)  that seems to be written for klipper but as it's based on the octoprint API, it should talk directly to marlin I think.

**Comment by gothcha** [comment]


**Reply by pmaersk** [reply]
And yet thanks again :-)

**Reply by gkr321** [reply]
Is the source of Q1 and Q3 connected to ground at the connection of Q2 right on the 4.22 board or the 4.27 board or are they different.

**Reply by gothcha** [reply]
Yes the sources of all three fets are connected to ground, it appears that the connection dot is missing on the 4.2.2 schematic. The only difference between the boards is that Creality re-numbered the parts!

I think it would be clearer if I use individual grounds for the Q1,Q2 & Q3, I will correct this & update the diagram.

**Reply by gothcha** [reply]
Latest version of schematics

[Creality 4.2.2 - Schematic 29-7-21.pdf](https://github.com/Jyers/Marlin/files/6899220/Creality.4.2.2.-.Schematic.29-7-21.pdf)
[Creality 4.2.7 - Schematic 29-7-21.pdf](https://github.com/Jyers/Marlin/files/6899222/Creality.4.2.7.-.Schematic.29-7-21.pdf)

**Comment by weaselpolarfrog** [comment]
Thanks for providing the schematic, it was extremely helpful.  To improve on it, I think I found a couple of minor issues with the schematic.

For the 4.2.2 schematic, it looks like R20-R23 should be pull-downs instead of pull-ups. I was probing them and was wondering why I didn't get continuity when I was verifying that I didn't short UART to 3.3V by ohm'ing between R19 and R23 on the far side of the stepper driver.  The pull-down theory is also reinforced that it coincides with the TMC2208 datasheet stating that CLK should be tied low (albeit without a pulldown resistor).

More of a nit than functional, the TVS diode, U8, symbol for the TVS is a bit off (the diodes are facing the wrong direction for both the GND and VCC. Refer to the U8 datasheet https://pdf1.alldatasheet.net/datasheet-pdf/view-marking/1294961/LEIDITECH/SR05L.html)
The board has it U8 oriented physically as
2 1
3 4
so you can just change the pin ordering of your symbol but leave the connections as is and swap the directions of the diodes and I think it will be fine.

Again, awesome work.

**Reply by gothcha** [reply]
@TOTA42 yes you are totally correct, I have the resistors correct on the 4.2.7. On the diodes I must admit I was a little lazy getting the symbol correct, as you say it doesn’t really affect the understanding, however I shall fix it up now you have found the correct part.  Thank you for checking this out and I am glad that the schematic has been useful.

**Comment by gothcha** [comment]
Update on the schematics to correct protection diode symbol and R21-R24 connection on 4.2.2 board. Thanks to @TOTA42


[Creality 4.2.7 - Schematic 22-9-21.pdf](https://github.com/Jyers/Marlin/files/7209530/Creality.4.2.7.-.Schematic.22-9-21.pdf)
[Creality 4.2.2 - Schematic 22-9-21.pdf](https://github.com/Jyers/Marlin/files/7209583/Creality.4.2.2.-.Schematic.22-9-21.pdf)

**Reply by DiskDoctor** [reply]
FANTASTIC! Thank you!

**Comment by Woodcat64** [comment]
I noticed on my board (v4.2.2) the p/n for mosfets (or at least the Q3 which has no heatsink) is HY1403, not HV1403.
Thanks for the drawings, very handy.

**Reply by gothcha** [reply]
You are correct, a typo, well spotted.  I will correct this when I next have the files open. Glad they have been useful.

**Comment by unixneo** [comment]
Very useful.  I was thinking to replace the flimsy micro usb port with something stronger and more resistant to "the slightest bump" when my wife sweeps away the stringies and dug out supports.   Thank you for this schematic!

**Comment by Woodcat64** [comment]
I think I found another one. The RX/TX pins on LCD port are swapped. It should be LCD port pin7 - TX, pin4 - RX. I wanted to connect to RPi using UART and it would not connect until I tried to swap the wires. I also verified this by looking at the connector on the display it self, which is labeled RX/TX so the port on motherboard must be TX/RX.

**Reply by gothcha** [reply]
A moot point whether the definition is relative to the micro or the LCD. I just used the pinout given for the LCD so as you say the TX refers to the LCD transmitting and RX is data to the LCD. I will probably leave it as in general it is not something most users would need to worry about. It is also always confusing on the FTDI boards as to which direction the TX/RX refers to!  Thank you for pointing this put however at least it is now clear.

**Comment by DragRedSim** [comment]
Just glancing at these schematics, I do have to question the U10 (HC245) chip on both. It appears that the channels of input and output are not connected correctly, as the schematics have CPU pin PA0 hooked to bus A5, PA1 to A6 and PA2 to A7. However, bus B7 is disconnected on the output side, meaning PA2 would not be driving anything; bus B6 is listed as being tied to the bed output, while B5 controls the nozzle heater, and the disconnected B4 is controlling the cooling fans. I suspect this is just a schematic error, either on one side or the other of the HC245 chip, given that all functions work fine on the actual board; simply that the internal connections of the HC245 chip are diagonally linked, rather than straight through.

**Reply by gothcha** [reply]
Yes you are correct, thanks for spotting it. I will update the schematics.

**Reply by pmaersk** [reply]
Hi Gotcha.

Do you have an update to the 22-9-21 version for 4.2.2 and 4.2.7 schematics for the U10 mention in the note?

Thx.

Peter Maersk-Moller

**Reply by gothcha** [reply]
Hi Peter.

I have done the corrections for you.

[Creality 4.2.7 - Schematic 28-5-22.pdf](https://github.com/Jyers/Marlin/files/8791328/Creality.4.2.7.-.Schematic.28-5-22.pdf)
[Creality 4.2.2 - Schematic 28-5-22.pdf](https://github.com/Jyers/Marlin/files/8791329/Creality.4.2.2.-.Schematic.28-5-22.pdf)

**Reply by pmaersk** [reply]
Thanks a million for your great work. Many here are very happy for your good work.
Currently hacking a bit and now soldering wires to unused pins on the CPU. It is physical challenging.

**Reply by gothcha** [reply]
Thank you - Good luck, I have done this too it is fiddly work!

**Reply by TruncatedPi** [reply]
Thankyou so much for sharing your work!
I found another minor schematic error with U10. Pin20 (Vcc) should be connected directly to +5V (and not to pin1). I verified this on my v4.2.2 board.

**Reply by gothcha** [reply]
Yes you are correct, I routed pin 20 (Vcc) to the wrong side of the resistor!  I shall include the correction the next time I have the schematics open.

**Reply by pmaersk** [reply]
Always appreciated. Thanks.

**Reply by lulol** [reply]
@gothcha For completeness, maybe you could add the schematic for the endstop switches PCB to your pdf when you release the pending bugfix update?

My stock Ender 3 Pro uses a 2-wire connection (V,S) for all 3 axis. The circuit shown below.

![creality_endstop_switch_sch](https://user-images.githubusercontent.com/14217781/184011220-590e5d44-0b2d-4de9-964e-d687c22def08.png)
![creality_endstop_switch](https://user-images.githubusercontent.com/14217781/184011235-78bdfe62-72e5-4e5a-8db1-72012e990d3d.jpg)

**Comment by kos-vkg** [comment]
Hello. I am an amateur.
Please tell me why in the marlin firmware for 4.2.7 boards are set by the driver 4955, although the board set 2225 (2208)?

**Reply by Cal-LF** [reply]
They are not set to 4955. Config file for E3V2 is 2208
![driver](https://user-images.githubusercontent.com/76499230/143953040-fe61eda9-55ff-4fdc-8443-cb2566a75723.jpg)
.

**Comment by lulol** [comment]
The stepper driver chips for Creality v4.2.x boards allegedly can be identified without needing to remove the heatsinks just looking at the hand written code on the SD card slot. [(source)](https://www.th3dstudio.com/hc/product-information/3rd-party-control-boards/creality-boards/creality-v4-2-x-board-driver-codes/)

### Creality V4.2.X Board Driver Codes

_The letters written on the card reader slot (with marker) are different depending on which stepper driver it has.
These are for the ones 4.2.X board with STM32F103RET6 controller:_

```
C = HR4988
E = A4988
A = TMC2208
B = TMC2209
H = TMC2225
```
![Creality v4 2 2_small](https://user-images.githubusercontent.com/14217781/172671556-e3ea6034-1a5c-4d1a-8bee-e1ca1be99404.jpg)

**Reply by MicroSur** [reply]
![image](https://user-images.githubusercontent.com/18687113/184170848-061a8a9d-407a-4106-a762-9045f5ee6988.png)

**Reply by sidorovav** [reply]
It looks like I have the same board. What is the controller chip?

**Reply by JPT77** [reply]
I've got a sticker saying T5 on my 427 board. What does that mean?

**Reply by gothcha** [reply]
I believe it indicates TMC2225 stepper driver chips

**Comment by DragRedSim** [comment]
Just found another pretty major error on the 4.2.7 schematic when toning out some connections to try and figure out what was going on with a board after a flash went wrong; the BOOT0 pin is listed as being pulled up to 3.3V via a resistor (R68, 10K). However, it is actually being pulled DOWN to ground. I assume it was a simple error caused by being right next to the non-polarised decoupling capacitor for the main processor.

**Reply by lulol** [reply]
You are right. BOOT0 needs to be low at boot otherwise the bootloader will assume that the flash is blank.

Also the 4.2.2 schematic has the same issue: here BOOT0 pin60 is connected to 3.3V via a 10K R59.

**Reply by gothcha** [reply]
Hi guys, yes your are correct - BOOT0 should be shown pulled LOW. I wouldn't however call it a major error but I will update it when I have a chance.

**Reply by pmaersk** [reply]
Thanks to all for continous updates :-) and special thanks to gotcha for schematics.

**Comment by noobyme** [comment]
Thank you, very cool! This saves me from learning about all electronics to see if it is possible to replace the drivers, I too wished too wish to replace the drivers on my ender 3 board, although for a different purpose as I have both a v4.2.2 and v4.2.7, I wish to use the old board to make my ender 3 multi material with raspberry pi to use 2 boards at the same time, inspired by me learning that voron uses 2 boards. I had only found a single post from a Russian man on reddit before this for the older ender boards. I think reddit would love to see this, don't you think? The more eyes the better. Also just to confirm before I make the order, you have already replaced the TMC2208/A4988 with TMC2209 and it worked completely fine without any other modifications? Thanks again.

**Comment by Clumpton** [comment]
Some attention needed for the wiring of K-Fan and K-Fan2. At present it is identical on both 4.2.2 4.2.7 diagrams.
![P1270644](https://user-images.githubusercontent.com/59048696/200830926-3c4483fd-3ad2-4327-a15c-47f5e1b8b433.JPG)

Fuse F3 does not appear to be present on the 4.2.7 board. On the 4.2.2 board it is physically between K-Fan1 socket and Q3 FET (where it is impossible to change without unsoldering everything).

So 4.2.7 either does not have F3 and relies on F1 for protection, or F3 is now hidden under the FET heatsinks. My 4.2.7 board is brand new, so I prefer not to remove them.

My 4.2.2 board definitely has F3 because I've blown it. However K-FAN1 still remains powered so must be connected to 24V before F3. 
On delivery the hot-end fan was connected to K-FAN2 and ran continuously, I can deduce that K-FAN2 -ve pin 1 is connected directly to ground on the 4.2.2 board that I have. I need to swap it out to check, but maybe someone else has one lying around? 
Thanks for the excellent work!

**Reply by vov32** [reply]
```
[heater_fan heatbreak_cooling_fan]
pin: PC0
heater_temp: 50.0
```
I tried to control heatsink fan through PC0, but there is 24v all the time on that pin.

**Reply by gothcha** [reply]
@Clumpton Sorry I didn't reply I have not been monitoring this for while. Yes you are correct the 4.2.7 board does not have fuse F3. The 4.2.2 board has both K-Fan1 and K-Fan2 switched with the Mosfet. The hot end fan should be connected to the permanent 24V Fan1 port at the bottom. If the -ve of your 4.2.2 is connected to ground this may be a fault?
I will update the schematic with the correction.

**Reply by gothcha** [reply]
@vov32 The K-Fan1 & K-Fan2 (part cooling and mainboard cooling) on the 4.2.2 and 4.2.7 are controlled from PA0 not PC0. This should be being used for the part cooling fan, there is no method of having an addition PWM controlled hot end fan on these boards. I also do not understand how you can detect 24V on PC0 which is a processor output pin and should be 3.3V max? On 4.2.7 boards I have used PC0 and PC1 as two of the UART controls for Linear advance.

**Comment by SiWoC** [comment]
![image](https://github.com/Jyers/Marlin/assets/12462639/5d577e8b-7647-4163-8651-3eccd63b8d45)
I see these 24V of Fan+, Bed+ and Noz+ appear to be directly connected.
But when I look on my PCB, a separate path leads to the Fan+ and when I try to measure, I get 24V on the Bed+ and Noz+, but not on the Fan+ (only some 2.3V, which looks like a defect on my board), so there seems to be something in between.
Anybody know what that would be and should it be added to the schema?

**Reply by Mackan2023** [reply]
Hello, have you started the fan and measured? because it is not active after the 4 layer if it is set to that.

**Reply by jovicassia** [reply]
great job, since finding the original schematic is **practically impossible**

Just one detail, the diagram is not the same as my PCB (it's a 4.2.2)

On my board, BED+ and NOZ+ are connected directly to the 24V input

FAN(+) / K-FAN1(+) / K-FAN2(+) = these 3 points are powered through fuse F3
it is the protection of the FANs circuit

In my case, the FAN wire touched the hotend, melted the cover and shorted out, burning F3

Then all the FANs stopped working

Since reaching fuse F3 to replace it is practically impossible, the solution was to install an external fuse holder connected between the 24V input and FAN+ (very easy to adapt) and if the fuse burns out, it is very easy to replace and maintains the protection

**Comment by Mackan2023** [comment]
![IMG_20240102_131500203](https://github.com/Jyers/Marlin/assets/153629228/9eb6fd00-ee34-48fc-a67f-b728aa5feec1)
could this be E = A4988 ?, it is 4.2.2 board with STM32F103RET6 on it. (3-pro)

**Reply by JJones780** [reply]
Looks more like a C=HR4988   ?
#warning "Creality 4.2.2 boards come with a variety of stepper drivers. Check the board label (typically on SD Card module) and set the correct *_DRIVER_TYPE! (C=HR4988, E=A4988, A=TMC2208, B=TMC2209, H=TMC2225, H8=HR4988). 
From src/inc/Warnings.cpp and shown during compiling.

**Comment by yw4z** [comment]
maybe its a good place to ask this question for 4.2.7 board. Can i connect a micro switch to BL touch socket and use it in klipper as klicky probe?

**Reply by JPT77** [reply]
As I  understood the z-end-stop uses the same line as the bl touch connector. so i fear you cannot use the bl touch for anything else while you need the z-end-stop. 
I could be wrong, though. at least the BL touch got more lines. 
I do not have a picture in mind about the 427 board now... but some boards do have additional pins for custom use?
I believe there was a full schematic of the board someone created by reverse-engineering. no idea where to find.

**Reply by yw4z** [reply]
schematics shared in here https://klipper.discourse.group/t/creality-board-4-2-2-and-4-2-7-schematics/3104
i'm not a electronics expert but it looks like they are not connected together

**Reply by yw4z** [reply]
but i guess original source is first post of this discussion

Not sure this one will work without frying my board but here is what i want
![1524a93d5b518c8840d9323f5fe81a7d35295789_2_1024x628](https://github.com/Jyers/Marlin/assets/28517890/4e490b22-7011-4f71-9bf0-b25362d32f1b)

**Reply by noobyme** [reply]
> maybe its a good place to ask this question for 4.2.7 board. Can i connect a micro switch to BL touch socket and use it in klipper as klicky probe?

I did that but with a klack ender microswitch probe instead. I used a jst 5 pin with only the right 2 sockets used. Could use dupont too I think. No it did not fry my board

**Reply by yw4z** [reply]
@noobyme oh nice to know its working. so you keep z end stop. do you use marlin or klipper. can you share your config file if you use klipper

**Reply by noobyme** [reply]
> @noobyme oh nice to know its working. so you keep z end stop. do you use marlin or klipper. can you share your config file if you use klipper

I used klipper but I didn't use jlas klicky probe. I eventually stopped using it because I got a bltouch clone. I had both a v4.2.2 and v4.2.7 board so I don't remember which board I used, I also remember trying to use both probe and z end stop at the same time but I'm not sure if I succeeded. I think I did though

Looking through klackender firmware, not sure how klicky probe works but it should be similar, all you have to do is define probe with your pin as PB1

[probe]
pin: ^YOUR_PIN #Change to where you plug your probe in, endstop or probe pin pulled high (^)
z_offset: 0 #Measure per your specific setup
x_offset: 4 # negative = left of the nozzle
y_offset: 21 # negative = in front of of the nozzle
speed: 5.0
lift_speed: 15.0
sample_retract_dist: 1
samples: 2
samples_tolerance_retries: 6

**Reply by yw4z** [reply]
@noobyme thanks

**Comment by bmourit** [comment]
I believe there is a problem with the PA11 PA12 and PB7 PB6. The schematic shows PA11 and PA12 using I2C for the EEPROM, however the actual code assigns PB7 and PB6 to those. Also PA6 and PA7 should be setup as SPI, but not so in the schematic. This is for board version 4.2.2

The documentation states that PA11 and PA12 are USB pins (but we obv don't have that on this board), but the pins can also be remapped for CAN peripheral or USART1 hardware flow control (which we also don't currently use, but could). Either way, they don't go to the EEPROM.

**Reply by gothcha** [reply]
Hi, I have double checked the boards and the schematic is definitely correct PA11/12 pins 44/45 are clearly connected to the I2C. Also  PA6 and PA7 are used for the Ymin and Zmin sensors. The code can be tricky as the definitions are overwritten depending on the exact board types so the references you are using may be incorrect. See photo for the connections of PA11/12 to the EEPROM.
Pin 48 is the top right in the photo.
<img width="693" alt="4 4 2 board PA11:12" src="https://github.com/user-attachments/assets/24df1a43-1cf7-4b2d-acf6-bfb2f3691e43">

**Comment by gothcha** [comment]
Also same for the 4.2.7 board
<img width="541" alt="4 2 7 PA11:12" src="https://github.com/user-attachments/assets/a6763910-4aa2-42ee-8731-025d7d6b1d49">

Also see photo for the 4.2.2 PA6/7 (pins 22/23 - 6 and 7 from left at bottom). These clearly go the Y and Z sensors.

<img width="536" alt="4 2 2 PA6:7" src="https://github.com/user-attachments/assets/ec3359c7-54b0-45d0-928a-1267215cbaa9">

**Comment by gothcha** [comment]
Just for completeness here is the pinout from the datasheet
<img width="1121" alt="STM32F103" src="https://github.com/user-attachments/assets/9db86860-3571-40d7-b034-31c4eed19b29">

**Comment by bmourit** [comment]
It was PB6 and PB7. The PA6/7 was hardware SPI support, but we don't use SPI on them. I think the EEPROM is using software i2c since these pins don't support hardware i2c, which solves all of this. The confusion was due to the software support, since I assumed it was Hardware I2C and since these pins don't support HW I2C, the manual also doesn't list them (I was going off the manual).

**Comment by Ferz82** [comment]
Hi. Can I use reality 4.2.2 with A4988 drivers?

**Reply by gothcha** [reply]
No they are not compatible

**Reply by lulol** [reply]
@Ferz82 Assuming you refer to Creality v4.2.2 board, you should use the drivers matching your board according to the code shown in [this comment](https://github.com/Jyers/Marlin/discussions/814#discussioncomment-2907513) above.

@gothcha my Creality v4.2.2 board has A4988 drivers installed and the Marlin firmware I compiled and installed has a `Configuration.h` config file with:
```
//#warning "Creality 4.2.2 boards may have A4988 or TMC2208_STANDALONE drivers. Check your board and make sure to select the correct DRIVER_TYPE!"

#define X_DRIVER_TYPE A4988
#define Y_DRIVER_TYPE A4988
#define Z_DRIVER_TYPE A4988
#define E0_DRIVER_TYPE A4988
```

**Comment by yw4z** [comment]
hi did i fried my board? D2 and D4 flashes when i try to start and it keeps restarting. it started after i replaced a faulty fan

**Reply by yw4z** [reply]
this one was power issue. caused by faulty fan cable. replacing cable fixed it

**Comment by kcduke42** [comment]
I apologize for asking in this thread, but does anyone by any chance have the schematic for the FFP0173 Aquila Main Board v1.0.1? I can't find it anywhere.

**Comment by NVSProgrammer** [comment]
You just saved my Ender 3 - Arduino UNO project!
